### PR TITLE
HOTT-1048: refactored page header handling

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

[HOTT-1048](https://transformuk.atlassian.net/browse/HOTT-1048)

### What?

I have added/removed/altered:

- [x] Generalised the page header handling

### Why?

I am doing this because:

- I'll want to use this on pages which don't have the existing search section on

### CODE REVIEW

File: spec/helpers/application_helper_spec.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@
       end
     end
 
-    context 'string contains Javascript code' do
+    context 'with string contains Javascript code' do
```
Comment (alexdesi) [inline spec/helpers/application_helper_spec.rb]: maybe it reads better with "when": 
"when string contains Javascript code"
("when string does not contains Javascript code")

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -27,6 +27,12 @@ def generate_breadcrumbs(current_page, previous_pages)
     tag.div(class: 'govuk-breadcrumbs') { tag.ol(crumbs.join('').html_safe, class: 'govuk-breadcrumbs__list', role: 'breadcrumbs') }
   end
 
+  def page_header(heading, &block)
+    extra_content = block_given? ? capture(&block) : nil
```
Comment (alexdesi) [inline app/helpers/application_helper.rb:31]: I've never used `capture`, but it seems appropriate for this case 👍

**Review by alexdesi — APPROVED** [review]
it looks good 👍

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (jebw) [reply app/helpers/application_helper.rb:31]: Its pretty much the exact use case it was designed for - its how a lot of the built in rails helpers work

File: spec/helpers/application_helper_spec.rb (hunk 1 above)
Comment (jebw) [reply spec/helpers/application_helper_spec.rb]: Yep, will amend

**Pull request merged**